### PR TITLE
CRS-1087 Retrieve recallLength from a bookings Fixed Term Recall - refactor and deprecate existing endpoint to cater for this

### PR DESCRIPTION
**Pull request by yasinmustafa**
As part of this I deprecated the `return-to-custody` end point because it wasn't extensible. Therefore, created a new endpoint (based on the old one)  and marked the old one as deprecated.

The old endpoint is only used by the calculate-release-dates service; therefore once it has been refactored to use the new endpoint I will revisit prison-api and remove the deprecated endpoint and associated code

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 1]
```diff
@@ -1260,9 +1261,20 @@ public Page<BedAssignment> getBedAssignmentsHistory(@PathVariable("bookingId") @
         @ApiResponse(responseCode = "400", description = "Invalid request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
         @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
         @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "Gets the return to custody date for a booking")
+    @Operation(summary = "Gets the return to custody date for a booking", description = "Do not use - replaced with /fixed-term-recalls", deprecated = true, hidden = true)
     @GetMapping("/{bookingId}/return-to-custody")
     public ReturnToCustodyDate getReturnToCustodyDate(@PathVariable("bookingId") @Parameter(description = "The offender booking linked to the return to custody date.", required = true) final Long bookingId) {
```
Comment (yasinmustafa) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:1266]: This endpoint and associated downstream code shall be removed  after calculate-release-dates has been changed to use the new endpoint

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/FixedTermRecallDetails.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+@EqualsAndHashCode
+public class FixedTermRecallDetails {
+
+    @Schema(description = "The booking id")
+    private Long bookingId;
+
+    @Schema(description = "The date the offender returned to custody")
+    private LocalDate returnToCustodyDate;
+
+    @Schema(description = "The date the offender returned to custody")
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/FixedTermRecallDetails.java]: copy and paste?

**Review by andymarke — DISMISSED** [review]
lgtm

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/FixedTermRecallDetails.java (hunk 2 above)
Comment (yasinmustafa) [reply src/main/java/uk/gov/justice/hmpps/prison/api/model/FixedTermRecallDetails.java]: good spot - fixed

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**